api/v1alpha1/node_pools.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NodePoolSpec describes a pool of nodes to allocate
+type NodePoolSpec struct {
+	// CloudID is the identifier of the O-Cloud that generated this request. The hardware
+	// manager may want to use this to tag the nodes in its database, and to generate
+	// statistics.
+	//
+	// +kubebuilder:validation:Required
+	CloudID string `json:"cloudID"`
+
+	// Location is the geographical location of the requested node.
+	//
+	// +kubebuilder:validation:Required
+	Location string `json:"location"`
+
+	Site string `json:"site"`
```

Comment:
Given that these two vars are related and talking about GeoLocation (and also used in ClusterSpec)....can we simply put in a `type GeoLocation struct` and share them between the two specs instead of them being at the top level?

Reply:
Added a LocationSpec struct.  Since 'Site' is mapped to a 'location' per Dell,  I made location optional.